src/containers/SubjectViewer.jsx

```diff
@@ -0,0 +1,252 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import SVGImage from '../components/SVGImage';
+import { Utility } from '../lib/Utility';
+
+import {
+  setRotation, setScaling, setTranslation,
+  setViewerState, SUBJECTVIEWER_STATE,
+} from '../ducks/subject-viewer';
+
+const INPUT_STATE = {
+  IDLE: 0,
+  ACTIVE: 1,
+}
+
+class SubjectViewer extends React.Component {
+  constructor(props) {
+    super(props);
+    
+    //HTML element refs.
+    this.section = undefined;
+    this.svg = undefined;
```

Comment:
I generally practise declaring refs in the constructor so people know what to expect later on in the code. This isn't technically essential, but I'd rather be explicit in spelling out the parts/elements/members/dangly bits of any component.